AG-16374 Add IDs to CTAs & forms for Google Analytics

JamesSwinton opened the pull request:
Adds IDs to elements and forms for tracking in Google Analytics events

JamesSwinton committed:
landing page cta ids

JamesSwinton committed:
Pricing page CTAs

JamesSwinton committed:
sync shared components

taktran commented on external/ag-website-shared/src/components/footer/Footer.tsx:
```diff
@@ -19,7 +19,11 @@ const MenuColumns = ({ footerItems }: { footerItems: FooterItem[] }) =>
                     <li key={`${title}_${name}`}>
-                        <a href={urlWithBaseUrl(url)} {...(newTab ? { target: '_blank', rel: 'noreferrer' } : {})}>
+                        <a
+                            id={`${name.toLowerCase().replace(' ', '-')}-nav`}
```
you can also use `github-slugger` here eg,https://github.com/ag-grid/ag-charts/blob/latest/packages/ag-charts-website/src/components/gallery/components/GallerySeriesLink.jsx#L17

taktran commented on external/ag-website-shared/src/components/site-header/HeaderNav.tsx:
```diff
@@ -74,6 +74,7 @@ const HeaderLinks = ({
                 return (
                     <li key={title.toLocaleLowerCase()} className={linkClasses}>
                         <a
+                            id={`${title.replace(' ', '-').toLowerCase()}-nav`}
```
Same here with `github-slugger`

JamesSwinton replied:
Fixed in Grid PR

taktran reviewed (approved):
Looks good. Just a suggestion about creating slugs

taktran committed:
Update .gitrepo parent sha

taktran committed:
git subrepo commit external/ag-website-shared

subrepo:
  subdir:   "external/ag-website-shared"
  merged:   "2d14c1304c"
upstream:
  origin:   "git@github.com:ag-grid/ag-website-shared.git"
  branch:   "latest"
  commit:   "c30d0e5b3e"
git-subrepo:
  version:  "0.4.9"
  origin:   "https://github.com/ingydotnet/git-subrepo"
  commit:   "30db3b8"

JamesSwinton commented:
LGTM
